nightfall-optimist/src/routes/proposer.mjs

```diff
@@ -357,10 +349,26 @@ router.post('/de-register', auth, async (req, res, next) => {
  *          $ref: '#/components/responses/InternalServerError'
  */
 router.post('/withdrawStake', auth, async (req, res, next) => {
+  const ethAddress = req.app.get('ethAddress');
+  const ethPrivateKey = req.app.get('ethPrivateKey');
+
   try {
-    const proposerContractInstance = await getContractInstance(PROPOSERS_CONTRACT_NAME);
-    const txDataToSign = await proposerContractInstance.methods.withdrawStake().encodeABI();
-    res.json({ txDataToSign });
+    // Recreate Proposer contract
+    const proposersContractInstance = await getContractInstance(PROPOSERS_CONTRACT_NAME);
+
+    // Withdraw proposer stake
+    const txDataToSign = await proposersContractInstance.methods.withdrawStake().encodeABI();
+    const tx = {
```

Comment:
I see the following block of code being repeated in this file, maybe we could create a method that does the job. Follows the repeated block:
```
    const tx = {
      from: ethAddress,
      to: proposersContractInstance.options.address,
      data: txDataToSign,
      gas: 8000000,
    };
    const signedTx = await web3.eth.accounts.signTransaction(tx, ethPrivateKey);
    const receipt = await web3.eth.sendSignedTransaction(signedTx.rawTransaction);
```